bitacross-worker/enclave-runtime/src/lib.rs

```diff
@@ -486,203 +476,6 @@ pub unsafe extern "C" fn migrate_shard(
 	sgx_status_t::SGX_SUCCESS
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn sync_parentchain(
-	blocks_to_sync: *const u8,
-	blocks_to_sync_size: usize,
-	events_to_sync: *const u8,
-	events_to_sync_size: usize,
-	events_proofs_to_sync: *const u8,
-	events_proofs_to_sync_size: usize,
-	parentchain_id: *const u8,
-	parentchain_id_size: u32,
-	immediate_import: c_int,
-) -> sgx_status_t {
-	if let Err(e) = sync_parentchain_internal(
-		blocks_to_sync,
-		blocks_to_sync_size,
-		events_to_sync,
-		events_to_sync_size,
-		events_proofs_to_sync,
-		events_proofs_to_sync_size,
-		parentchain_id,
-		parentchain_id_size,
-		immediate_import == 1,
-	) {
-		error!("Error synching parentchain: {:?}", e);
-		return sgx_status_t::SGX_ERROR_UNEXPECTED
-	}
-
-	sgx_status_t::SGX_SUCCESS
-}
-
-#[allow(clippy::too_many_arguments)]
-unsafe fn sync_parentchain_internal(
-	blocks_to_sync: *const u8,
-	blocks_to_sync_size: usize,
-	events_to_sync: *const u8,
-	events_to_sync_size: usize,
-	events_proofs_to_sync: *const u8,
-	events_proofs_to_sync_size: usize,
-	parentchain_id: *const u8,
-	parentchain_id_size: u32,
-	immediate_import: bool,
-) -> Result<()> {
-	let blocks_to_sync = Vec::<SignedBlock>::decode_raw(blocks_to_sync, blocks_to_sync_size)?;
-	let events_to_sync = Vec::<Vec<u8>>::decode_raw(events_to_sync, events_to_sync_size)?;
-	let events_proofs_to_sync =
-		Vec::<StorageProof>::decode_raw(events_proofs_to_sync, events_proofs_to_sync_size)?;
-	let parentchain_id = ParentchainId::decode_raw(parentchain_id, parentchain_id_size as usize)?;
-
-	if !events_proofs_to_sync.is_empty() {
-		let blocks_to_sync_merkle_roots: Vec<sp_core::H256> =
-			blocks_to_sync.iter().map(|block| block.block.header.state_root).collect();
-		// fixme: vulnerability! https://github.com/integritee-network/worker/issues/1518
-		// until fixed properly, we deactivate the panic upon error altogether in the scope of #1547
-		if let Err(e) = validate_events(&events_proofs_to_sync, &blocks_to_sync_merkle_roots) {
-			warn!("ignoring event validation error {:?}", e);
-			//	return e.into()
-		}
-	}
-	dispatch_parentchain_blocks_for_import(
-		blocks_to_sync,
-		events_to_sync,
-		&parentchain_id,
-		immediate_import,
-	)
-}
-
-#[no_mangle]
-pub unsafe extern "C" fn ignore_parentchain_block_import_validation_until(
-	until: *const u32,
-) -> sgx_status_t {
-	let va = match get_validator_accessor_from_integritee_solo_or_parachain() {
-		Ok(r) => r,
-		Err(e) => {
-			error!("Can't get validator accessor: {:?}", e);
-			return sgx_status_t::SGX_ERROR_UNEXPECTED
-		},
-	};
-
-	let _ = va.execute_mut_on_validator(|v| v.set_ignore_validation_until(*until));
-
-	sgx_status_t::SGX_SUCCESS
-}
-
-/// Dispatch the parentchain blocks for import.
-/// Depending on the worker mode, a different dispatcher is used:
-///
-/// * An immediate dispatcher will immediately import any parentchain blocks and execute
-///   the corresponding extrinsics (offchain-worker executor).
```

Comment:
The doc mentions `offchain-worer` - this mode is currently used by bitacross worker. Are you sure it's safe to delete this ?

Reply:
it seems I've deleted some useful functionality. I was pretty sure that `sidechain.rs` file is referencing to only code which is irrelevant for bc-worker, but seems it just changes the context. Rolling it back.

what do you think, would it make sense to rename file to `offchain.rs`?

Reply:
I prefer to keep the original file name - less problems while applying upstream changes. Please add comment why file is kept.